src/index.ts

```diff
@@ -188,18 +189,32 @@ async function catchupAndWatchChain(
         return undefined;
       }
 
-      const res = await fetch(`${config.ipfsGateway}/ipfs/${cid}`, {
-        timeout: 2000,
-        retry: { retries: 10, maxTimeout: 60 * 1000 },
-        // IPFS data is immutable, we can rely entirely on the cache when present
-        cache: "force-cache",
-        cachePath:
-          config.cacheDir !== null
-            ? path.join(config.cacheDir, "ipfs")
-            : undefined,
-      });
-
-      return (await res.json()) as T;
+      const url = `${config.ipfsGateway}/ipfs/${cid}`;
+
+      chainLogger.trace(`Fetching ${url}`);
+
+      try {
+        const res = await fetch(url, {
+          timeout: 2000,
+          onRetry(cause) {
+            chainLogger.debug(`Retrying IPFS request ${String(cause)}`);
```

Comment:
You could preserve more information about the error with something like:

```ts
chainLogger.debug({ msg: `Retrying IPFS request`, err: cause })
```

That should make the stack trace available in the log.

Reply:
done, normally the URL is on the error but added the URL as well in the log just in case